torch/csrc/jit/script/init.cpp

```diff
@@ -161,6 +210,14 @@ struct ModuleValue : public SugaredValue {
   std::shared_ptr<Module> module;
 };
 
+static std::shared_ptr<SugaredValue> createConstantSugaredValue(py::object obj) {
+  if(py::isinstance<Module>(obj)) {
+    auto r = py::cast<std::shared_ptr<Module>>(obj);
+    return std::make_shared<ModuleValue>(r);
+  }
```

Comment:
Why do we need the special case for `Module`? Also, it won't make the module immutable, like it does with other Python objects.

NB `Module` here is the C++ module (i.e. only `ScriptedModule` in Python), right?

Reply:
Calls from ScriptModule->ScriptModule need to be handled differently than calls from ScriptModule->python module. So that is what is happening here.

Reply:
Sorry I didn't get that. Can you please expand your comment?

Reply:
This special case will be moved into the handling for nn.ModuleList. We need to make script modules into ModuleValues and normal modules into PythonValues because we need to ensure calls to script modules do not go through PythonValues. Otherwise, instead of inlining the graph, it would insert a python op into the graph to call the subgraph.